examples/sphere/shallow_water.jl

```diff
@@ -18,46 +19,154 @@ import OrdinaryDiffEq
 using OrdinaryDiffEq: ODEProblem, solve, SSPRK33
 
 using Logging
-usempi = get(ENV, "CLIMACORE_DISTRIBUTED", "") == "MPI"
 if usempi
     using ClimaComms
     using ClimaCommsMPI
-    const context = ClimaCommsMPI.MPICommsContext()
-    const pid, nprocs = ClimaComms.init(context)
-
-    # log output only from root process
-    logger_stream = ClimaComms.iamroot(context) ? stderr : devnull
-
-    prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
-    atexit() do
-        global_logger(prev_logger)
-    end
 else
     import TerminalLoggers
-    global_logger(TerminalLoggers.TerminalLogger())
-    context = nothing
 end
+using ClimaCorePlots
+import Plots
+
+# Physical parameters needed
+struct Params{FT}
+    R::FT
+    Ω::FT
+    g::FT
+    D₄::FT
+end
+Params(::Type{FT}) where {FT} = Params{FT}(
+    FT(6.37122e6),
+    FT(7.292e-5),
+    FT(9.80616),
+    FT(1.0e16), # hyperdiffusion coefficient
+)
```

Comment:
```suggestion
Base.@kwdef struct Params{FT}
    R::FT = FT(6.37122e6)
    Ω::FT = FT(7.292e-5)
    g::FT = FT(9.80616)
    D₄::FT = FT(1.0e16) # hyperdiffusion coefficient
end
```
I'd probably use `Base.@kwdef` here to reduce repetition, and then we can call this with `Params{FT}()`. We can do the same with some of the `AbstractTest` subtypes.

Reply:
Hopefully, `Base.@kwdef` will be documented in public domain, by next Julia release!